sda-commons-server-spring-data-mongo/src/test/java/org/sdase/commons/server/spring/data/mongo/SpringDataMongoWithValidationIT.java

```diff
@@ -0,0 +1,92 @@
+package org.sdase.commons.server.spring.data.mongo;
+
+import static io.dropwizard.testing.ConfigOverride.config;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.dropwizard.testing.junit5.DropwizardAppExtension;
+import java.time.LocalDate;
+import java.util.Optional;
+import javax.validation.ConstraintViolationException;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.sdase.commons.server.mongo.testing.MongoDbClassExtension;
+import org.sdase.commons.server.spring.data.mongo.example.MyAppWithValidation;
+import org.sdase.commons.server.spring.data.mongo.example.MyConfiguration;
+import org.sdase.commons.server.spring.data.mongo.example.model.Person;
+import org.sdase.commons.server.spring.data.mongo.example.model.PhoneNumber;
+import org.sdase.commons.server.spring.data.mongo.example.repository.PersonRepository;
+import org.springframework.data.mongodb.core.MongoOperations;
+
+class SpringDataMongoWithValidationIT {
+
+  @RegisterExtension
+  @Order(0)
+  static final MongoDbClassExtension mongo = MongoDbClassExtension.builder().build();
+
+  @RegisterExtension
+  @Order(1)
+  static final DropwizardAppExtension<MyConfiguration> DW =
+      new DropwizardAppExtension<>(
+          MyAppWithValidation.class,
+          null,
+          config("springDataMongo.connectionString", mongo::getConnectionString));
+
+  @Test
+  void shouldStartup() {
+    assertThat((MyAppWithValidation) DW.getApplication()).isNotNull();
+  }
+
+  @Test
+  void shouldValidateNameAndAge() {
+    MyAppWithValidation app = DW.getApplication();
+    MongoOperations mongoOperations = app.getMongoOperations();
+    var validationError =
+        Assertions.assertThrows(
+            ConstraintViolationException.class, () -> mongoOperations.save(createPerson(0, null)));
+
+    assertThat(validationError.getMessage()).contains("age: must be greater than or equal to 1");
+    assertThat(validationError.getMessage()).contains("name: must not be null");
```

Comment:
Java offers internationalization as a "feature". The test fails on my machine because I see the following error message:
```
name: darf nicht null sein, age: muss größer-gleich 1 sein
```

Reply:
There is [`@DefaultLocale`](https://github.com/junit-pioneer/junit-pioneer/blob/main/src/main/java/org/junitpioneer/jupiter/DefaultLocale.java). It's worth a try although I saw it working in some cases and failing in others.

Reply:
I validated the properties of the constraint validation exception. This way we can validate the property name and the constraint without having to worry about the message.